### DISCUSSION TITLE
Question about determinstic blockchain and linkchain

### DISCUSSION THREAD

**Discussion by medhananda** [Cyfrin/foundry-full-course-cu#995] [Q&A]
I am not understanding the difference between these two.
I understand that if the nodes make calls to API, they might get different results at different times and hence won't work in blockchain.
However, when I look at chain link feeds e.g. price feed that too will be different at different times and so different nodes may get different feeds, e.g. forex changes all the time. I guess I am looking for more clarity about how the chain link feeds maintain the deterministic nature of the blockchains.

**Comment by 0xusmanf** [comment]
> However, when I look at chain link feeds e.g. price feed that too will be different at different times and so different nodes may get different feeds, e.g. forex changes all the time. I guess I am looking for more clarity about how the chain link feeds maintain the deterministic nature of the blockchains.

Whenever you request a price feed from a chainlink priceFeed contract, it gives you an already stored on-chain static price value which is why every node requesting the priceFeed will get the same value.

How often the chainlink contract updates the on-chain stored price value is dependent on the two variables defined below.

#### Primary Aggregation Parameter: Deviation Threshold
The deviation threshold indicates when the feed will update. In the ETH/USD case, when the off-chain price, as reported by the data aggregators’ APIs, deviates 0.5% from the on-chain trusted price, the feed updates. Each Chainlink deviation threshold is customizable depending on the smart contract needs.
#### Secondary Aggregation Parameter: Heartbeat
Heartbeat is a countdown timer that updates the price on-chain when it reaches 00:00. It is a backup in the event that the deviation threshold does not trigger an update over the length of the heartbeat. If assets have very low price volatility, they may not reach deviation thresholds often, so the heartbeat ensures data feeds remain fresh.

You see these two variables for every price feed when you click on 
- [x] Show more details

like in the example below.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110072406/8654354b-c83c-4b03-a261-11903f3b8ce8)

You can find more about how Chainlink price feeds work in [this blog post](https://blog.chain.link/analyze-decentralized-oracles-in-real-time/).
